zaza/model.py

```diff
@@ -1045,6 +1045,35 @@ async def async_block_until_all_units_idle(model_name=None, timeout=2700):
 block_until_all_units_idle = sync_wrapper(async_block_until_all_units_idle)
 
 
+async def async_block_until_unit_count(application, target_count,
+                                       model_name=None, timeout=2700):
+    """Block until the number of units matches target_count.
+
+    An example accessing this function via its sync wrapper::
+
+        block_until_unit_count('keystone', 4)
+
+    :param application_name: Name of application
+    :type application_name: str
+    :param target_count: Number of expected units.
+    :type target_count: int
+    :param model_name: Name of model to interact with.
+    :type model_name: str
+    :param timeout: Time to wait for status to be achieved
+    :type timeout: float
+    """
+    async def _check_unit():
+        model_status = await async_get_status()
+        unit_count = len(model_status.applications[application]['units'])
+        return unit_count == int(target_count)
```

Comment:
Given that we've documented that `target_count` needs to be an int, I'd rather have a check that this argument is an int (with a RuntimeError if not) and then stop casting it to an int in the `async_block_until`